jetbrains-core/tst/software/aws/toolkits/jetbrains/services/s3/S3VirtualBucketTest.kt

```diff
@@ -197,4 +200,16 @@ class S3VirtualBucketTest {
         assertThat(request.prefix()).isEqualTo("prefix/")
         assertThat(request.delimiter()).isEqualTo("/")
     }
+
+    @Test
+    fun getUrl() {
+        val s3Client = mockClientManager.create<S3Client>()
+
+        s3Client.stub {
+            on { utilities() } doReturn S3Utilities.builder().region(Region.US_WEST_2).build()
+        }
+
+        val sut = S3VirtualBucket(Bucket.builder().name("TestBucket").build(), s3Client)
+        assertThat(sut.generateUrl("prefix/key")).isEqualTo(URL("https://s3.us-west-2.amazonaws.com/TestBucket/prefix/key"))
```

Comment:
what about the virtual-hosted path style?

Reply:
We have to enable that on our S3 client and we don't do that at all today.